### PR TITLE
Updating Trigger Path and Fixing for Empty Histograms for single lept…

### PR DESCRIPTION
**Pull request by soureek**
Adding path HLT_Ele25_WPTight_Gsf_v in the list of paths to be monitored. Modification of topHLTriggerOfflineDQM sequence to solve the empty monitoring histogram for single lepton paths issue.

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

**Comment by Soureek89** [comment]
Please merge this PR as soon as possible if everything is OK.

File: DQMOffline/Trigger/src/TopSingleLeptonHLTOfflineDQM.cc [hunk 1]
```diff
@@ -612,17 +692,21 @@ namespace HLTOfflineDQMTopSingleLepton {
...
             fill("leptDeltaREta_", isoMuons[mIndMatched]->eta(), mDeltaRMin);   
             if (lMatched) fill("leptResolution_", fabs(isoMuons[mIndMatched]->pt()-muonRefs_[0]->pt())/isoMuons[mIndMatched]->pt() );   
           }
-          if (lMatched) fill("matchingMon_", 0.5 );
+          
+		  if (lMatched) fill("matchingMon_", 0.5 );
           else isMatched = false;
-          if (j1IndMatched < 500) {
+          
+		  if (j1IndMatched < 500) {
```
Comment (dmitrijus) [inline DQMOffline/Trigger/src/TopSingleLeptonHLTOfflineDQM.cc]: Two tabs followed by two spaces!

File: DQMOffline/Trigger/src/TopDiLeptonHLTOfflineDQM.cc [hunk 2]
```diff
@@ -349,7 +353,7 @@ namespace HLTOfflineDQMTopDiLepton {
         }
       }
       fill("jetMult_", mult);
-
+	  std::cout<<"Test DiLepton 4"<<std::endl;
```
Comment (dmitrijus) [inline DQMOffline/Trigger/src/TopDiLeptonHLTOfflineDQM.cc]: Please remove this, and other uses of thread-unsafe std::cout.
Or move to MessageLogger / LogDebug.

File: DQMOffline/Trigger/src/TopSingleLeptonHLTOfflineDQM.cc (hunk 1 above)
Comment (Soureek89) [inline DQMOffline/Trigger/src/TopSingleLeptonHLTOfflineDQM.cc]: Removed tab spaces.

File: DQMOffline/Trigger/src/TopDiLeptonHLTOfflineDQM.cc (hunk 2 above)
Comment (Soureek89) [inline DQMOffline/Trigger/src/TopDiLeptonHLTOfflineDQM.cc]: Removed cout statements.

**Comment by Soureek89** [comment]
Please check the latest commit.

**Comment by Soureek89** [comment]
Please process this PR at your earliest.

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Comment by davidlange6** [comment]
@Soureek89 @dmitrijus - this PR is making a huge noise in prompt reco. I'm reverting for now.

the worst part is

%MSG
%MSG-w TopSingleLeptonHLTOfflineDQM:   TopSingleLeptonHLTOfflineDQM:SingleTopSingleMuonHLTOfflineDQM  03-Aug-2016 22:52:14 CEST Run: 277069 Event: 54472424
No RAW trigger summary found! Returning... 

%MSG
%MSG-w TopSingleLeptonHLTOfflineDQM:   TopSingleLeptonHLTOfflineDQM:SingleTopSingleMuonHLTOfflineDQM  03-Aug-2016 22:52:14 CEST Run: 277069 Event: 54472424
No AOD trigger summary found! Returning... 

%MSG
%MSG-w TopSingleLeptonHLTOfflineDQM:   TopSingleLeptonHLTOfflineDQM:SingleTopSingleMuonHLTOfflineDQM  03-Aug-2016 22:52:14 CEST Run: 277069 Event: 54114308
No RAW trigger summary found! Returning... 

%MSG
%MSG-w TopSingleLeptonHLTOfflineDQM:   TopSingleLeptonHLTOfflineDQM:SingleTopSingleMuonHLTOfflineDQM  03-Aug-2016 22:52:14 CEST Run: 277069 Event: 54114308
No AOD trigger summary found! Returning... 

%MSG
%MSG-w TopSingleLeptonHLTOfflineDQM:   TopSingleLeptonHLTOfflineDQM:SingleTopSingleMuonHLTOfflineDQM  03-Aug-2016 22:52:14 CEST Run: 277069 Event: 54114308
No RAW trigger summary found! Returning... 

%MSG
%MSG-w TopSingleLeptonHLTOfflineDQM:   TopSingleLeptonHLTOfflineDQM:SingleTopSingleMuonHLTOfflineDQM  03-Aug-2016 22:52:14 CEST Run: 277069 Event: 54114308
No AOD trigger summary found! Returning...

**Comment by Soureek89** [comment]
@davidlange6 @dmitrijus - This warning message appears if the input file does not contain either of the input tags "hltTriggerSummaryRaw" and "hltTriggerSummaryAOD" [1],[2]. These are needed to collect hlt objects and match them with reco objects for topHLTOfflineDQM. I am not aware if anything regarding the label or type of the above mentioned input tags have changed in between. If yes, I have to update the code.    

[1]https://github.com/Soureek89/cmssw/blob/topHLTDQM_Update_Jun15/DQMOffline/Trigger/src/TopSingleLeptonHLTOfflineDQM.cc#L157
[2]https://github.com/Soureek89/cmssw/blob/topHLTDQM_Update_Jun15/DQMOffline/Trigger/src/TopSingleLeptonHLTOfflineDQM.cc#L509

**Comment by davidlange6** [comment]
maybe this module is running on data sets that it should not be?

but an example file I have has

triggerTriggerEvent_hltTriggerSummaryAOD__HLT. 26264.5 6916.36

but still generates the error.




On Aug 4, 2016, at 11:31 AM, Soureek Mitra <notifications@github.com<mailto:notifications@github.com>> wrote:


@davidlange6<https://github.com/davidlange6> @dmitrijus<https://github.com/dmitrijus> - This warning message appears if the input file does not contain either of the input tags "hltTriggerSummaryRaw" and "hltTriggerSummaryAOD" [1],[2]. These are needed to collect hlt objects and match them with reco objects for topHLTOfflineDQM. I am not aware if anything regarding the label or type of the above mentioned input tags have changed in between. If yes, I have to update the code.

[1]https://github.com/Soureek89/cmssw/blob/topHLTDQM_Update_Jun15/DQMOffline/Trigger/src/TopSingleLeptonHLTOfflineDQM.cc#L157
[2]https://github.com/Soureek89/cmssw/blob/topHLTDQM_Update_Jun15/DQMOffline/Trigger/src/TopSingleLeptonHLTOfflineDQM.cc#L509

**Comment by Soureek89** [comment]
This module in under "topHLTriggerOfflineDQM" sequence defined in [1]. "topHLTriggerOfflineDQM" is then added to the more generic sequence "offlineHLTSource" defined in [2]. "offlineHLTSource" is termed as "triggerOfflineDQMSource" in [3] which in turn is called during generic offline DQM process in [4]. 

[1]https://github.com/Soureek89/cmssw/blob/topHLTDQM_Update_Jun15/DQMOffline/Trigger/python/topHLTOfflineDQM_cff.py#L9
[2]https://github.com/Soureek89/cmssw/blob/topHLTDQM_Update_Jun15/DQMOffline/Trigger/python/DQMOffline_Trigger_cff.py#L63
[3]https://github.com/Soureek89/cmssw/blob/topHLTDQM_Update_Jun15/DQMOffline/Trigger/python/DQMOffline_Trigger_cff.py#L81 
[4]https://github.com/Soureek89/cmssw/blob/topHLTDQM_Update_Jun15/DQMOffline/Configuration/python/DQMOffline_cff.py#L56

**Pull request merged**